subprojects/docs/src/docs/userguide/authoring-builds/basics/writing_settings_files.adoc

```diff
@@ -0,0 +1,308 @@
+
+The settings file can optionally define the plugins your project uses with link:{javadocPath}/org/gradle/plugin/management/PluginManagementSpec.html[`pluginManagement`], including binary repositories such as the Gradle Plugin Portal or other Gradle builds using `includeBuild`:
+
+[source]
+----
+pluginManagement {
+    repositories {
+        gradlePluginPortal()
+        google()
+    }
+    includeBuild("../my-build-logic")
+}
+----
+
+You can also include plugins and plugin dependency strategies in this block.
+
+=== 2. Apply shared plugins
+
+The settings file can optionally define the link:{javadocPath}/org/gradle/plugin/use/PluginDependenciesSpec.html[plugins], which are used as shared configurations among several builds / subprojects:
+
+[source]
+----
+plugins {
+    id("org.gradle.toolchains.fake") version "0.6.0"
+}
+----
+
+This is typically used to ensure that all subprojects use the same plugin version.
+
+=== 3. Define the root project name
+
+The settings file defines your project name using the link:{javadocPath}/org/gradle/api/initialization/ProjectDescriptor.html[`rootProject.name` property]:
+
+[source]
+----
+rootProject.name = "root-project"
+----
+
+There is only one root per project.
+
+=== 4. Add components and dependencies
+
+The settings file can optionally link:{javadocPath}/org/gradle/api/initialization/resolve/DependencyResolutionManagement.html[define] the locations of components your project relies on (as well as "how-to" resolve them) using `repositories` such as binary repositories like Maven Central and/or other Gradle builds using `includeBuild`:
+
+[source]
+----
+dependencyResolutionManagement {
+    repositories {
+        mavenCentral()
+    }
+    includeBuild("../other-project")
+}
+----
+
+You can also include version catalogs in this section.
+
+=== 5. Add subprojects to the build
+
+The settings file defines the structure of the project by adding all the subprojects using the link:{javadocPath}/org/gradle/api/initialization/Settings.html[`include`] statement:
+
+[source]
+----
+include("app")
+include("business-logic")
+include("data-model")
+----
+
+== Settings File Scripting
+
+There are many more properties and methods on the `Settings` object that you can use to configure your build.
+
+It's important to remember that while many Gradle scripts are typically written in short Groovy or Kotlin syntax, every item in the settings script is essentially invoking a method on the `Settings` object in the Gradle API:
+
+[source]
+----
+include("app")
+----
+
+Is actually:
+
+[source]
+----
+settings.include("app")
+----
+
+Additionally, the full power of the Groovy and Kotlin languages is available to you.
+
+For example, instead of using `include` many times to add subprojects, you can iterate over the list of directories in the project root folder and include them automatically:
+
+[source]
+----
+rootDir.listFiles().filter { it.name!="gradle" && it.isDirectory && !it.isHidden }.forEach {
```

Comment:
This script seems fragile. I don't think we should be recommending it. 
Instead, maybe something like 

```suggestion
rootDir.listFiles().filter { it.isDirectory && (new File(it, "build.gradle.kts").exists()) }.forEach {
```